File: csharp-dotnet2/src/main/CsharpDotNet2/estuary-client/Model/TypesIpfsPin.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace estuary-client.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class TypesIpfsPin {
    /// <summary>
    /// Gets or Sets Cid
    /// </summary>
    [DataMember(Name="cid", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "cid")]
    public string Cid { get; set; }

    /// <summary>
    /// Gets or Sets Meta
    /// </summary>
    [DataMember(Name="meta", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "meta")]
    public Object Meta { get; set; }

    /// <summary>
    /// Gets or Sets Name
    /// </summary>
    [DataMember(Name="name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "name")]
    public string Name { get; set; }

    /// <summary>
    /// Gets or Sets Origins
    /// </summary>
    [DataMember(Name="origins", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "origins")]
    public List<string> Origins { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class TypesIpfsPin {\n");
      sb.Append("  Cid: ").Append(Cid).Append("\n");
      sb.Append("  Meta: ").Append(Meta).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  Origins: ").Append(Origins).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/lib/model/types_ipfs_pin.dart
part of swagger.api;

class TypesIpfsPin {
  
  String cid = null;
  

  Object meta = null;
  

  String name = null;
  

  List<String> origins = [];
  
  TypesIpfsPin();

  @override
  String toString() {
    return 'TypesIpfsPin[cid=$cid, meta=$meta, name=$name, origins=$origins, ]';
  }

  TypesIpfsPin.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    cid =
        json['cid']
    ;
    meta =
      
      
      new Object.fromJson(json['meta'])
;
    name =
        json['name']
    ;
    origins =
        (json['origins'] as List).map((item) => item as String).toList()
    ;
  }

  Map<String, dynamic> toJson() {
    return {
      'cid': cid,
      'meta': meta,
      'name': name,
      'origins': origins
     };
  }

  static List<TypesIpfsPin> listFromJson(List<dynamic> json) {
    return json == null ? new List<TypesIpfsPin>() : json.map((value) => new TypesIpfsPin.fromJson(value)).toList();
  }

  static Map<String, TypesIpfsPin> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, TypesIpfsPin>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new TypesIpfsPin.fromJson(value));
    }
    return map;
  }
}

